### DISCUSSION TITLE
focus last over monitors

### DISCUSSION THREAD

**Discussion by d3m1gd** [YaLTeR/niri#479] [Ideas]
`focus-workspace-previous` and `workspace-auto-back-and-forth` seem to work within a single monitor, and it would be nice to have actions that span monitors as well, as in `focus-workspace-previous-over-monitors` and `workspace-auto-back-and-forth-over-monitors` so that with a single key press we colud jump to last workspace on another monitor and back-and-forth would be jumping back across monitors as well.

for example. monitor A has named workspaces "1" "2" "3", and B has "4" "5" "6". 
assume we are on "5", and back-and-forth is active, so focusing 2 and then again focusing 2 will focus "1" or "3" which is surprising, if one expects to land back on "5". and `focus-workspace-previous-over-monitors` is a simple action allowing to Alt-Tab between monitors with ease. They are in the same idea thread, because they carry a same idea behind, of switching between monitors with repeated keypress of a single key.

`focus-workspace-previous-over-monitors` does not mean to focus other monitor in every instance, but actual last workspace. in example above with A and B, if we are on "5" and previous was "6", the action is supposed to select "6", but if previous was "2", then jump to "2", keeping history list across monitors.

**Comment by YaLTeR** [comment]
I think it'd make sense if `focus-workspace-previous` and `workspace-auto-back-and-forth` worked across monitors too. Might've been an omission when named workspaces were implemented. (Would need some refactoring though.) @algernon do you remember if there was a specific reason they weren't changed?

**Reply by algernon** [reply]
No specific reason other than me forgetting it.

**Comment by TyberiusPrime** [comment]
I had a brief foray into the source...
What would be the right 'id' to store for the previous monitor?
OutputId isn't copy, so 'trivially' expanding previous_workspace_idx to be a tuple won't work without cloning.

**Comment by Moelf** [comment]
since this is a discussion, I will also just ask: is there a workaround of any sort right now?

**Reply by TyberiusPrime** [reply]
I don't have one. You could watch the event stream and do the decisions in external code, but I don't have anything handy right now.
